custom_components/sat/climate.py

```
@@ -789,8 +753,12 @@ async def _async_control_setpoint(self, pwm_state: PWMState) -> None:
             _LOGGER.info("Running PWM cycle with state: %s", pwm_state)
 
             if pwm_state == PWMState.ON and self.max_error > -DEADBAND:
-                self._setpoint = self.minimum_setpoint
-                _LOGGER.debug("Setting setpoint to minimum: %.1f°C", self.minimum_setpoint)
+                if not self._dynamic_minimum_setpoint:
+                    self._setpoint = self._coordinator.minimum_setpoint
+                else:
+                    self._setpoint = self.adjusted_minimum_setpoint - BOILER_TEMPERATURE_OFFSET
```

Ruff (0.8.2)

`BOILER_TEMPERATURE_OFFSET` may be undefined, or defined from star imports

(F405)